library/ssl_tls.c

```diff
@@ -1403,24 +1403,10 @@ static int ssl_populate_transform( mbedtls_ssl_context *ssl )
 #endif /* MBEDTLS_CIPHER_MODE_CBC */
 
 
+    /* Initialize Zlib contexts */
 #if defined(MBEDTLS_ZLIB_SUPPORT)
-    // Initialize compression
-    //
     if( session->compression == MBEDTLS_SSL_COMPRESS_DEFLATE )
```

Comment:
Ok, I'll try to expand the commit message when I rewrite history.

Reply:
Manuel left for holiday, I tried to spot the rationale with this change, but it isn't very obvious so I don't think it is worth the effort to figure out the explanation for the change.

Reply:
The compression feature isn't implemented in accordance with the current SSL transform architecture: One the one hand, one might argue that the SSL transforms are responsible for record protection/deprotection, not for compression/decompression, because `ssl_[de]compress_buf()` is not called from `ssl_[de/en]crypt_buf()`. On the other hand, one might argue that they are responsible for compression/decompression, because `mbedtls_ssl_transform` contains the compression context. 

If compression is treated as part of SSL transforms, it is arguable whether temporary compression buffers should be part of the transform or not -- the current architecture has no precedent for this, because `ssl_[de/en]crypt_buf()` operates in-place. If it is part of the transform, allocation should happen in `ssl_populate_transform()`, but then the buffer should be referenced from `mbedtls_ssl_transform`, not `mbedtls_ssl_context`. If the compression buffer is not part of the transform, allocation should happen outside of `ssl_populate_transform()`, e.g. in `derive_keys()` as done here.

In any case, thinking about this reveals that compression doesn't fit the architecture in many places, and given that it's deprecated and will be removed soon, I don't think it's worth making it fit or explaining any changes related to it in too much detail.

Reply:
I'd prefer a little more something in the commit message than what's there now, but given that this matches what was merged into baremetal already, OK.